genDocker.go

```
@@ -0,0 +1,227 @@
+	dockerfile = append(dockerfile, "RUN gem install bundler && bundle install")
+	dockerfile = append(dockerfile, "COPY . /app")
+	dockerfile = append(dockerfile, "EXPOSE $PORT")
+
+	if writeStringFile(path.Join(dir, "Dockerfile"), strings.Join(dockerfile, "\n")) == false {
+		panic(fmt.Sprintf("Failed to write Dockerfile for %s", dir))
+	}
+}
+
+func genSinatraDocker(dir string, conf ServiceConfig, compose *[]string) {
+	appendDockerComposeService(compose, dir, conf.Port, "bundle exec ruby server.rb -o '0.0.0.0'", true)
+
+	var dockerfile []string
+	dockerfile = append(dockerfile, "FROM ruby:2.6")
+	dockerfile = append(dockerfile, "WORKDIR /app")
+	dockerfile = append(dockerfile, "COPY Gemfile /app/Gemfile")
+	dockerfile = append(dockerfile, "COPY Gemfile.lock /app/Gemfile.lock")
+	dockerfile = append(dockerfile, "RUN gem install bundler && bundle install")
+	dockerfile = append(dockerfile, "COPY . /app")
+	dockerfile = append(dockerfile, "EXPOSE $PORT")
+
+	if writeStringFile(path.Join(dir, "Dockerfile"), strings.Join(dockerfile, "\n")) == false {
+		panic(fmt.Sprintf("Failed to write Dockerfile for %s", dir))
+	}
+}
+
+func updatePGConf() {
+	// Idempotent.
+	// Clone the behavior from these links so postgres is open to other hosts:
+	// https://github.com/docker-library/postgres/blob/87b15b6c65ba985ac958e7b35ba787422113066e/11/Dockerfile#L160
+	// https://github.com/docker-library/postgres/blob/818958f40191c614b1e17dc9e5249ddbb7406a60/11/alpine/docker-entrypoint.sh#L119
+	pgconf := path.Join(".local-compose", "postgres", "data", "postgresql.conf")
+	input, err := ioutil.ReadFile(pgconf)
+	if err == nil {
+		lines := strings.Split(string(input), "\n")
+
+		for i, line := range lines {
+			if strings.Contains(line, "#listen_addresses = 'localhost'") {
+				lines[i] = "listen_addresses = '*'"
+			}
+		}
+		if writeStringFile(pgconf, strings.Join(lines, "\n")) == false {
+			panic("Failed to write postgresql.conf")
+		}
+	}
+
```

I would maybe fail here if ReadFile fails for a reason other than "file not found".